### PR TITLE
release: Prepare for 0.3.1

### PR DESCRIPTION
**Pull request by real-or-random**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,20 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [0.3.1] - 2023-04-10
+We strongly recommend updating to 0.3.1 if you use or plan to use Clang >=14 to compile libsecp256k1, e.g., Xcode >=14 on macOS has Clang >=14. When in doubt, check the Clang version using `clang -v`.
+
+#### Security
+ - Fix "constant-timeness" issue with Clang >=14 that could leave applications using libsecp256k1 vulnerable a timing side-channel attack. The fix avoids secret-dependent control flow and secret-dependent memory accesses in conditional moves of memory objects when libsecp256k1 is compiled with Clang >=14.
```
[the remark this answers is not in the document]
Comment (real-or-random) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,20 @@ All notable changes to this project will be documented in this file.
...
+We strongly recommend updating to 0.3.1 if you use or plan to use Clang >=14 to compile libsecp256k1, e.g., Xcode >=14 on macOS has Clang >=14. When in doubt, check the Clang version using `clang -v`.
+
+#### Security
+ - Fix "constant-timeness" issue with Clang >=14 that could leave applications using libsecp256k1 vulnerable to a timing side-channel attack. The fix avoids secret-dependent control flow and secret-dependent memory accesses in conditional moves of memory objects when libsecp256k1 is compiled with Clang >=14.
+
+#### Added
+  - Added tests against [Project Wycheproof's](https://github.com/google/wycheproof/) set of ECDSA test vectors (Bitcoin "low-S" variant), a fixed set of test cases designed to trigger various edge cases.
+
+#### Changed
+ - Increased minimum required CMake version to 1.13. CMake builds remain experimental.
```
[the remark this answers is not in the document]
Comment (real-or-random) [reply CHANGELOG.md]: sure.. -.- fixed.

File: configure.ac [hunk 3]
```diff
@@ -6,15 +6,15 @@ AC_PREREQ([2.60])
...
+define(_PKG_VERSION_IS_RELEASE, true)
 
 # The library version is based on libtool versioning of the ABI. The set of
 # rules for updating the version can be found here:
 # https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
 # All changes in experimental modules are treated as if they don't affect the
 # interface and therefore only increase the revision.
 define(_LIB_VERSION_CURRENT, 2)
-define(_LIB_VERSION_REVISION, 1)
+define(_LIB_VERSION_REVISION, 2)
```
Comment (jonasnick) [inline configure.ac:17]: Why increase the lib version?

Comment (real-or-random) [reply configure.ac:17]: indeed.

**Comment by real-or-random** [comment]
forced-push to rebase on master, just to make sure

File: configure.ac (hunk 3 above)
Comment (jonasnick) [reply configure.ac:17]: Maybe I'm missing something obvious, but this library version was never released. I'm not sure why we would treat the library version different to the package version in that regard. The lib version was incremented in [#1228](https://github.com/bitcoin-core/secp256k1/pull/1228/commits/28e63f7ea75af50af0bfde34309993f276bbc0ae) at the same time when the package version was updated to 0.3.1. We do this so that master has a different version than the previous release.

**Review by jonasnick — COMMENTED** [review]
ACK 898e1c676e177af73b2c7b540cc93e72a2790fa2

**Comment by real-or-random** [comment]
Let's maybe wait for CI.

**Comment by real-or-random** [comment]
see also https://github.com/bitcoin-core/secp256k1/pull/1268#issuecomment-1501997547, please. I assume it's okay to merge this PR here without the "[Unreleased]" section, but I can add it if you insist, of course.

**Comment by hebasto** [comment]
FWIW, [here](https://github.com/bitcoin-core/secp256k1/issues/1136#issuecomment-1509741749) is an ABI compatibility report created with [ABI Compliance Checker](https://github.com/lvc/abi-compliance-checker).

**Review by alokeutpal — APPROVED** [review]
